cache.go

```diff
@@ -161,39 +174,65 @@ func (c *Cache) Delete(key string) bool {
 	return false
 }
 
-// Clears the cache
+// Clears the cache.
+// This is a control command.
 func (c *Cache) Clear() {
 	done := make(chan struct{})
 	c.control <- clear{done: done}
 	<-done
 }
 
-// Stops the background worker. Operations performed on the cache after Stop
-// is called are likely to panic
+// Stops the background worker.
+// This is a control command.
 func (c *Cache) Stop() {
-	close(c.promotables)
-	<-c.control
+	go func() {
+		time.Sleep(c.stopDelay)
```

Comment:
I'm not sure what's gained by this delay. Your comment on the other issue (which would probably be good to copy into the description of this PR) said it's "to let concurrent work finish", but what would happen if the work did not finish? That is, what is the bad consequence of still having some stuff in the other channels when the worker notices the `stop` message in the control channel and shuts down? It will drain those channels anyway, so this doesn't seem to be about reducing the chance of leaving some values sitting around in the channels.

As far as I can tell, the worst that would happen is that it wouldn't update the LRU status of some recently updated items. And since LRU status won't be getting updated from this point on anyway, I would think it'd be adequate to say that calling Stop() implies expiration/eviction behavior is now undefined.

Maybe in place of the deleted text in the comment above, it could say something like "After Stop is called, Get and Set operations will not cause panics, but the state of the cache data is no longer guaranteed to be consistent and should not be relied on, and control operations will have no effect.